sdk/storage/storage-file-datalake/review/storage-file-datalake.api.md

```diff
@@ -0,0 +1,1424 @@
+    move(destinationPath: string, options?: PathMoveOptions): Promise<PathMoveResponse>;
+    move(destinationFileSystem: string, destinationPath: string, options?: PathMoveOptions): Promise<PathMoveResponse>;
+    readonly name: string;
+    setAccessControl(acl: PathAccessControlItem[], options?: PathSetAccessControlOptions): Promise<PathSetAccessControlResponse>;
+    setHttpHeaders(httpHeaders: PathHttpHeaders, options?: PathSetHttpHeadersOptions): Promise<PathSetHttpHeadersResponse>;
+    setMetadata(metadata?: Metadata, options?: PathSetMetadataOptions): Promise<PathSetMetadataResponse>;
+    setPermissions(permissions: PathPermissions, options?: PathSetPermissionsOptions): Promise<PathSetAccessControlResponse>;
+    toDirectoryClient(): DataLakeDirectoryClient;
+    toFileClient(): DataLakeFileClient;
+}
+
+// @public (undocumented)
+export interface DataLakeRequestConditions extends ModifiedAccessConditions, LeaseAccessConditions {
+}
+
+// @public
+export class DataLakeSASPermissions {
+    add: boolean;
+    create: boolean;
+    delete: boolean;
+    static parse(permissions: string): DataLakeSASPermissions;
+    read: boolean;
+    toString(): string;
+    write: boolean;
+}
+
+// @public
+export interface DataLakeSASSignatureValues {
+    cacheControl?: string;
+    contentDisposition?: string;
+    contentEncoding?: string;
+    contentLanguage?: string;
+    contentType?: string;
+    expiresOn?: Date;
+    fileSystemName: string;
+    identifier?: string;
+    ipRange?: SasIPRange;
+    pathName?: string;
+    permissions?: DataLakeSASPermissions;
+    protocol?: SASProtocol;
+    snapshotTime?: string;
+    startsOn?: Date;
+    version?: string;
+}
+
+// @public
+export class DataLakeServiceClient extends StorageClient {
+    constructor(url: string, credential?: StorageSharedKeyCredential | AnonymousCredential | TokenCredential, options?: StoragePipelineOptions);
+    constructor(url: string, pipeline: Pipeline);
+    getFileSystemClient(fileSystemName: string): DataLakeFileSystemClient;
+    getUserDelegationKey(startsOn: Date, expiresOn: Date, options?: ServiceGetUserDelegationKeyOptions): Promise<ServiceGetUserDelegationKeyResponse>;
+    listFileSystems(options?: ServiceListFileSystemsOptions): PagedAsyncIterableIterator<FileSystemItem, ServiceListFileSystemsSegmentResponse>;
+}
+
+export { deserializationPolicy }
+
+// @public
+export interface DirectoryCreateOptions extends PathCreateOptions {
+}
+
+// @public (undocumented)
+export interface DirectoryCreateResponse extends PathCreateResponse {
+}
+
+// @public (undocumented)
+export interface FileAppendOptions extends CommonOptions {
+    // (undocumented)
+    abortSignal?: AbortSignalLike;
+    // (undocumented)
+    conditions?: LeaseAccessConditions;
+    // (undocumented)
+    onProgress?: (progress: TransferProgressEvent) => void;
+    // (undocumented)
+    transactionalContentMD5?: Uint8Array;
```

Comment:
can someone explain `transactional` in this context?

Reply:
Well, it's kind of storage naming history for the hash value used to check data integrity in one simple unit upload operation, like append to append blob, or stage block. In blob, we have transactionalContentMD5 and transactionalContentCrc64.